### ISSUE TITLE
suggestion: remove use of options("oceTz")

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2109]
I ran across this in working on #2108, the first comment of which noted that the docs on `tz` might need improvement.  I agree.  My guess is that the statement (Roxygen format)
```R
#' @param tz time zone.  The value `oceTz` is set at package setup.
```
might not mean much to users.  I can change that to
```R
#' @param tz time zone.  The value obtained with `options()$oceTz` is used, if not supplied.
```

But I'm guessing that a fair number of users don't know how to set options.   Also, this is non-reproducible; for another user to get the same results you have to supply R code, data files, *and* a line from your `~/.Rprofile` file.

The following indicates that we set `tz` as `options()$oceTz` in only a fairly small number of cases.

I propose changing to default `UTC`, because then consulting e.g. `?read.rsk` will reveal the default.

This might break some user code, but (a) I doubt that a lot of users are setting non-UTC timezones in `~/.Rprofile` and (b) any user with data in a non-UTC zone is likely specifying `tz=` in the function call.

So, I'm asking for a co-developer vote, from @richardsc and @clayton33.  I hope @avictor88 and  @cameronrichrdson will offer opinions also, as new users of oce.

Please vote
* -1 to retain the current action (that some, but not all, `tz` values default to `options()$oceTz`)
* +1 to change so that `tz` defaults to `"UTC"` for all functions
* any number in between to indicate fractional preference (e.g. voting 0 means "no opinion").

I will sum the votes and make a decision by the upcoming Saturday (June 24) at noon local time.

**Comment by clayton33** [comment]
+1 from me !

**Comment by dankelley** [comment]
+1 from me !

**Comment by dankelley** [comment]
After thinking some more about this, and chatting with @richardsc, I think the best plan is to document the option more clearly, so my vote is now -1.

However, I'm going to leave this open for a while because I need to think more about `tz` as a function parameter, generally.  Clark and I discussed this at length today.  It's all a bit confusing.  Another factor is that I think R changed on this in recent years.  It used to be -- I think -- that `as.POSIXct(t, tz="something new")` did not change the numerical value you get if you print the object.  But now it does change that.  (I used to use `lubridate::with_tz()` to get what I wanted, but now `as.POSIXct()` does what I want.)

I think I like the following (new) scheme:
* if `tz` is not provided, try to find that info in the file (which it almost never will find, if our examination of a bunch of CNV files is accurate) and, if there is nothing there, use UTC (or perhaps use the `oceTz` value, as before).
* if `tz` is given as a string, use that. 

A big question is what to do if the provided value contradicts what is in the file.  Do we let the file decide or the user?  Do we report contradictions?

There are other considerations, too.

I know that rsk files can have an indication of timezone *offset* if you hook up a phone/tablet to the machine for logging.  But they seem to have nothing about a timezone otherwise.  And it doesn't see that CNV or ODF files have timezones, either --although that's from just some quick tests during a meeting today.  I don't think adp and adv files have zone information, either.  In summary: I can't think of any cases in which the file has timezone information.  That means that the user ought always to decide.  It's only the user, probably looking at some hand-written notes, who will know how the instrument was set up.  So maybe the user should be tasked with deciding on the zone.  For that, having the code assume UTC and then letting the user change later seems like a reasonable option.

I'm going in circles.  And I've been in these circles before.  It's a can of worms, made worse by the age of the functions (and thus a desire not to change them).

I just made a vignette about package options.  I think I ought to write one about times.  And *that* might be the place to clear things up, with examples.

Below are some notes on how things work.  (These could be cleaned up and explained in a vignette that tells users how to convert from UTC to another zone.)

```R
> t <- Sys.time()
> t
[1] "2023-06-22 17:40:28.397 ADT"
> as.numeric(t)
[1] 1687466428
> 
> t2 <- lubridate::with_tz(t, "UTC")
> t2
[1] "2023-06-22 20:40:28.397 UTC"
> as.numeric(t2)
[1] 1687466428
> 
> t3 <- as.POSIXct(t, "UTC")
> t3
[1] "2023-06-22 20:40:28.397 UTC"
> as.numeric(t3)
[1] 1687466428
> 
> t4 <- t
> attr(t4, "tzone") <- "UTC"
> t4
[1] "2023-06-22 20:40:28.397 UTC"
> as.numeric(t4)
[1] 1687466428
```
